### PR TITLE
Check for hostError and automaticRestart when test finishes.

### PR DESCRIPTION
**Pull request by mborsz**
**What type of PR is this?**
/kind flake

**What this PR does / why we need it**:
When test finishes, check gce activity logs for hostError and automaticRestart. This is useful for debugging failed (flaky) test attempts.
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: cluster/log-dump/log-dump.sh [hunk 1]
```diff
@@ -429,6 +429,30 @@ function dump_nodes_with_logexporter() {
...
+                              --format='value(creationTimestamp)')
+    echo "Failures for ${group}"
+    gcloud logging read --order=asc \
+          --format='value(timestamp,jsonPayload.resource.name,jsonPayload.event_subtype)' \
+          --project "${PROJECT}" \
+          "resource.type=\"gce_instance\"
+           logName=\"projects/${PROJECT}/logs/compute.googleapis.com%2Factivity_log\"
+           (jsonPayload.event_subtype=\"compute.instances.hostError\" OR jsonPayload.event_subtype=\"compute.instances.automaticRestart\")
+           jsonPayload.resource.name:\"${group}\"
+           timestamp >= \"${creation_timestamp}\""
```
Comment (wojtek-t) [inline cluster/log-dump/log-dump.sh]: This looks great.
My only question here is - what is the estimation on the time it takes to run this command? This pretty much goes over activity log on all nodes, which for 5k-node clusters may be quite a lot.
Did you try running that on larger clusters?

Comment (mborsz) [reply cluster/log-dump/log-dump.sh]: It takes ~20 seconds per instance group. So approx. 1 minute for all of them.

Comment (wojtek-t) [reply cluster/log-dump/log-dump.sh]: Cool - thanks.

**Comment by mikedanese** [comment]
We started to see LogDump failures in GKE when this merged:

https://gubernator.k8s.io/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gke/34578

Could this be related?

**Comment by mborsz** [comment]
It is related, log says:

```
W1130 16:04:52.986] Zone: us-central1-f
W1130 16:04:54.206] INSTANCE_GROUPS=
W1130 16:04:54.206] NODE_NAMES=
W1130 16:04:54.206] ./cluster/log-dump/log-dump.sh: line 438: INSTANCE_GROUPS[@]: unbound variable
```

where line 438 is added in this PR. Will revert and check what the problem is next week.

**Pull request merged**